### PR TITLE
Bz#1535271 externalauth ad

### PR DESCRIPTION
**Pull request by dayleparker**
Added new sections from Kbase article: "4.2. Configuring Authentication with Active Directory" and "4.4. Troubleshooting External Authentication Configuration", updated beta labels to 4.6 GA and a few other minor updates.

### CODE REVIEW

**Comment by dayleparker** [comment]
Hi Suyog,

Thank you for volunteering to review this addition!
Feedback welcome on the two new sections, general structure, etc.
Please see https://bugzilla.redhat.com/show_bug.cgi?id=1535271 for details and a preview.

Many thanks,
Dayle

File: doc-Managing_Authentication/topics/External_AD.adoc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+. Restart and enable the `httpd` service:
++
+----
+# systemctl restart httpd
+# systemctl enable httpd
+----
+
+Complete authentication setup by configuring the following from the {product-title_short} user interface:
+
+. Log in to the web interface as an administrative user.
```
Comment (suyogsainkar) [inline doc-Managing_Authentication/topics/External_AD.adoc]: user interface?

File: doc-Managing_Authentication/topics/External_AD.adoc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+. Select a *Session Timeout* if required.
+. Select *External (httpd)* as the authentication *Mode*.
+. Select *Enable Single Sign-On* to allow single sign-on using Kerberos tickets from client machines that authenticate to the same Active Directory domain as the appliance.
+. In the *Role Settings* area, select *Get User Groups from External Authentication (httpd)*.
+. Click *Save*.
+//add screenshot from article.
+
+[IMPORTANT]
+======
+* The configuration steps for the {product-title_short} user interface must be completed on each user interface and web services enabled appliance (appliances with `user_interface` or `web_services` server roles enabled).
```
Comment (suyogsainkar) [inline doc-Managing_Authentication/topics/External_AD.adoc]: I think "for the {product-title_short} user interface" can be eliminated here? > The configuration steps must be completed on each user interface and web services enabled appliance (appliances with `user_interface` or `web_services` server roles enabled).

File: doc-Managing_Authentication/topics/External_auth_troubleshooting.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
+[[external_troubleshooting]]
+= Troubleshooting External Authentication Configuration
+
+////
+ Is this needed? When do you need this?
+Using the appliance_console to generate External HTTP Authentication generates the following files:
+
+/etc/httpd/conf.d/manageiq-external-auth.conf
+/etc/sssd/sssd.conf
+With the above configuration files in place, HTTP will query, using dbus, SSSD which will perform the LDAP/S authentication and authorization steps necessary to validate users and groups.
```
Comment (suyogsainkar) [inline doc-Managing_Authentication/topics/External_auth_troubleshooting.adoc]: A quick online search shows me more instances of "D-Bus". Thought not sure if there's one correct way  - perhaps worth checking the IBM style guide?

File: doc-Managing_Authentication/topics/External_AD.adoc [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+//source content https://access.redhat.com/solutions/2751431
+
+This procedure outlines how to configure {product-title_short} to authenticate against an existing Active Directory (AD) configuration using external HTTP authentication. This provides Active Directory users access to the {product-title_short} appliance user interface, as well as the REST API.
+
+
+// EDIT A SUMMARY FOR THIS To enable external authentication using Active Directory complete the following steps to configure your HTTP server, then your {product-title_short} appliance.
+
+// Do we need Network Manager step from MIQ docs still?)
+
+. Connect to the {product-title_short} using SSH.
```
Comment (suyogsainkar) [inline doc-Managing_Authentication/topics/External_AD.adoc]: Add the word - appliance? Connect to the {product-title_short} appliance using SSH.

File: doc-Managing_Authentication/topics/External_AD.adoc (hunk 1 above)
Comment (dayleparker) [reply doc-Managing_Authentication/topics/External_AD.adoc]: Fixed.

File: doc-Managing_Authentication/topics/External_AD.adoc (hunk 2 above)
Comment (dayleparker) [reply doc-Managing_Authentication/topics/External_AD.adoc]: I've rearranged this a little and moved this idea to above the procedure, and have removed some of those repetitive words.

File: doc-Managing_Authentication/topics/External_auth_troubleshooting.adoc (hunk 3 above)
Comment (dayleparker) [reply doc-Managing_Authentication/topics/External_auth_troubleshooting.adoc]: Not in IBM, but Googling makes me agree with D-Bus. I've changed this (though it is commented out as I wasn't sure if it was needed)

File: doc-Managing_Authentication/topics/External_AD.adoc (hunk 4 above)
Comment (dayleparker) [reply doc-Managing_Authentication/topics/External_AD.adoc]: Much better - fixed.

**Comment by dayleparker** [comment]
Hi Suyog,
Thanks for your great feedback. I've fixed these issues, and in particular, did a bit of rearranging with the "Important" box and the part about running the steps on the web_services and user_interface.
Let me know what you think!
Cheers,
Dayle

**Comment by suyogsainkar** [comment]
@dayleparker Thanks Dayle, looks great. Merging to master.

**Pull request merged**